rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -294,6 +294,41 @@ TEST_F(TestExecutor, spin_some_elapsed) {
   ASSERT_TRUE(timer_called);
 }
 
+TEST_F(TestExecutor, spin_for_duration)
+{
+  DummyExecutor dummy;
+  auto node = std::make_shared<rclcpp::Node>("node", "ns");
+  bool timer_called = false;
+  auto timer =
+    node->create_wall_timer(
+    std::chrono::milliseconds(10), [&]() {
+      std::this_thread::sleep_for(std::chrono::milliseconds(10));
+      timer_called = true;
+    });
+  dummy.add_node(node);
+  // Wait for the wall timer to have expired.
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
+  dummy.spin_for(std::chrono::milliseconds(1));
+
+  ASSERT_TRUE(timer_called);
+}
+
+TEST_F(TestExecutor, spin_for_longer_timer)
+{
+  DummyExecutor dummy;
+  auto node = std::make_shared<rclcpp::Node>("node", "ns");
+  bool timer_called = false;
+  auto timer =
+    node->create_wall_timer(
+    std::chrono::milliseconds(10), [&]() {
+      timer_called = true;
+    });
+  dummy.add_node(node);
+  dummy.spin_for(std::chrono::milliseconds(5));
+
+  ASSERT_FALSE(timer_called);
```

Comment:
this test seems very prone to sporadic failures.
I would really like that unit-test didn't depend on wall time, but currently they do.
this means that if it takes more than 5ms for the executor to start spinning (due to a CPU spike in the CI server) this test will fail.

we should at least use larger intervals: for example a timer with 1s period.

Reply:
agree, this could be flaky.

Reply:
Done - I guess for the negative test where `spinning` takes less time than the `timer` - it should not be flaky (broken) anymore.